DashKit/DashPumpManager.swift

```diff
@@ -0,0 +1,911 @@
+//
+//  DashPumpManager.swift
+//  DashKit
+//
+//  Created by Pete Schwamb on 4/18/19.
+//  Copyright © 2019 Tidepool. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import LoopKit
+import os.log
+import PodSDK
+
+
+public protocol PodStatusObserver: class {
+    func didUpdatePodStatus()
+}
+
+public class DashPumpManager: PumpManager {
+
+    public static var managerIdentifier = "OmnipodDash"
+
+    var podCommManager: PodCommManagerProtocol
+
+    public let log = OSLog(category: "DashPumpManager")
+
+    public static let localizedTitle = LocalizedString("Omnipod DASH", comment: "Generic title of the omnipod DASH pump manager")
+
+    public var lastReconciliation: Date? {
+        // TODO
+        return Date()
+    }
+
+    public func roundToSupportedBasalRate(unitsPerHour: Double) -> Double {
+         return supportedBasalRates.filter({$0 <= unitsPerHour}).max() ?? 0
+    }
+
+    public func roundToSupportedBolusVolume(units: Double) -> Double {
+        return supportedBolusVolumes.filter({$0 <= units}).max() ?? 0
+    }
+
+    public var supportedBolusVolumes: [Double] {
+        // 0.05 units for rates between 0.05-30U
+        return (1...600).map { Double($0) / Double(Pod.pulsesPerUnit) }
+    }
```

Comment:
This method, for better or worse, is relying on the precision of division vs the rounding errors of addition. For the 12th element of the array, 12 / 20.0 = 0.6, which is correct.  But using the stride method, we get 0.55 + 0.05 = 0.6000000000000001. You'll see a lot of dose math in Loop using division instead of multiplication for this same reason.